tools/ibc-test/src/tests/ibc/sudt_erc20_transfer.rs

```diff
@@ -0,0 +1,577 @@
+use std::{collections::HashMap, process::Command};
+
+use bytes::Bytes;
+use ckb_ics_axon::message::Envelope;
+use ckb_sdk::{
+    constants::{SIGHASH_TYPE_HASH, TYPE_ID_CODE_HASH},
+    traits::{
+        CellCollector, CellQueryOptions, DefaultCellCollector, DefaultCellDepResolver,
+        DefaultHeaderDepResolver, DefaultTransactionDependencyProvider, SecpCkbRawKeySigner,
+        ValueRangeOption,
+    },
+    tx_builder::{balance_tx_capacity, unlock_tx, CapacityBalancer},
+    unlock::{ScriptUnlocker, SecpSighashUnlocker},
+    CkbRpcClient, ScriptId,
+};
+use ckb_types::{
+    core::{BlockView, Capacity, ScriptHashType, TransactionBuilder, TransactionView},
+    packed,
+    prelude::*,
+};
+use forcerelay_ckb_sdk::{
+    config::AddressOrScript,
+    search::{get_latest_cell_by_type_script, IbcChannelCell, PacketCell},
+    transaction::{
+        add_ibc_envelope, assemble_consume_ack_packet_partial_transaction,
+        assemble_send_packet_partial_transaction, assemble_write_ack_partial_transaction,
+    },
+};
+use futures::TryStreamExt;
+use ibc_test_framework::{
+    bootstrap::binary::channel::{bootstrap_channel_with_connection, BootstrapChannelOptions},
+    prelude::*,
+    relayer::axon::transfer::read_deployed_contracts,
+};
+use prost::Message;
+use tokio::runtime::Runtime;
+
+use crate::{
+    consts::CLIENT_TYPE_ARGS,
+    generator::{get_lock_script, PRIVKEY},
+    tests::ckb::packet::utils::search_sudt_cells,
+};
+
+/// Test CKB SUDT (source) <-> Axon ERC20 (sink).
+///
+/// Chain A must be ckb, and chain B must be axon.
+pub struct SudtErc20TransferTest;
+
+impl TestOverrides for SudtErc20TransferTest {}
+
+impl BinaryConnectionTest for SudtErc20TransferTest {
```

Comment:
Is this possible to encapsule these code into [this trait](https://github.com/synapseweb3/forcerelay/blob/2a9eed11258e86c79249dbd621a7acc738e7539e/tools/test-framework/src/chain/ext/transfer.rs#L77)?

Reply:
We can refactor this later for a generic transfer test. For now ckb as sink zone is not supported so the only practically supported scenario is CKB source SUDT to/from axon voucher ERC20.

Reply:
related discussion: https://github.com/synapseweb3/forcerelay/pull/332#discussion_r1338012102